### ISSUE TITLE
noUnusedLocals incorrectly reports `using` resources as unused unless explicitly used

### ISSUE THREAD

**Issue by dj-foxxy** [microsoft/TypeScript#61890]
### 🔎 Search Terms

noUnusedLocals
using

### 🕗 Version & Regression Information

Present on all releases supporting `using`

### ⏯ Playground Link

https://www.typescriptlang.org/play/?noUnusedLocals=true&target=99&ts=5.9.0-dev.20250618#code/N4KABBYK4M4JYDsDmYBmB7dYC8ZhgG0BlATwFsAjdAGwDoATOGAB3RgFMBdACgEo8AvmAEBuEAKA

### 💻 Code

```ts
{
    using foo = { [Symbol.dispose]() {} };
}
```

### 🙁 Actual behavior

`foo` is flagged as unused by `noUnusedLocals`

### 🙂 Expected behavior

`foo` is not flagged by `noUnusedLocals` because it's implicitly used at the end of the scope when its `[Symbol.dispose]` property is called.

### Additional information about the issue

Not explicitly referencing resources is common (e.g., mutexs) so this feels like a bug.

**Comment by MartinJohns** [comment]
Duplicate of #55538. Used search terms: `using unused in:title`

Just prefix your variable with `_` making it clear that this intentional.
